src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts

```diff
@@ -0,0 +1,269 @@
+    if (criteria === this.SCORE_VALUE) {
+      this.updateScoreFormControls();
+    } else if (criteria === this.CHOICE_CHOSEN_VALUE) {
+      this.updateChoiceFormControls();
+    }
+  }
+
+  private updateScoreFormControls(): void {
+    // use set timeout to give the form time to populate the path form group controls
+    setTimeout(() => {
+      const scores = this.branchPaths.map((branchPath) => branchPath.scores);
+      if (scores.length > 0) {
+        for (let i = 0; i < scores.length; i++) {
+          this.pathFormGroup.controls[`path-${i + 1}`].setValue(scores[i].join(', '));
+        }
+      }
+    });
+  }
+
+  private updateChoiceFormControls(): void {
+    // use set timeout to give the form time to populate the path form group controls
+    setTimeout(() => {
+      const choices = this.branchPaths.map((branchPath) => branchPath.choiceId);
+      if (choices.length > 0) {
+        for (let i = 0; i < choices.length; i++) {
+          this.pathFormGroup.controls[`path-${i + 1}`].setValue(choices[i]);
+        }
+      }
+    });
+  }
+
+  protected removeBranchButtonClicked(): void {
+    if (
+      confirm(
+        $localize`Are you sure you want to remove this branch?\n\nThe branch structure will be removed but the steps will not be deleted.`
+      )
+    ) {
+      this.removeBranch();
+    }
+  }
+
+  private removeBranch(): void {
+    for (let bp = 0; bp < this.branchPaths.length; bp++) {
+      const branchPath = this.branchPaths[bp];
+      this.removeBranchPath(branchPath);
+      bp--; // shift the counter back one because we have just removed a branch path
+    }
+
+    const nodeIdAfter = this.projectService.getNodeIdAfter(this.targetId);
+
+    /*
+     * update the transition of this step to point to the next step
+     * in the project. this may be different than the next step
+     * if it was still the branch point.
+     */
+    this.projectService.setTransition(this.targetId, nodeIdAfter);
+    this.projectService.setTransitionLogicField(
+      this.targetId,
+      'howToChooseAmongAvailablePaths',
+      null
+    );
+    this.projectService.setTransitionLogicField(this.targetId, 'whenToChoosePath', null);
+    this.projectService.setTransitionLogicField(this.targetId, 'canChangePath', null);
+    this.projectService.setTransitionLogicField(this.targetId, 'maxPathsVisitable', null);
+    this.projectService.calculateNodeNumbers();
+    this.saveProject();
+  }
+
+  /**
+   * Remove a branch path by removing all the branch path taken constraints
+   * from the steps in the branch path, resetting the transitions in the
+   * steps in the branch path, and removing the transition corresponding to
+   * the branch path in this branch point node.
+   * @param branch the branch object
+   */
+  protected removeBranchPath(branch: any): void {
+    const nodesInBranchPath = branch.nodesInBranchPath;
+    for (const checkedItem of nodesInBranchPath) {
+      const nodeId = checkedItem.nodeId;
```

Comment:
Simplify, and rename checkedItem -> nodeInBranchPath?
```suggestion
    for (const nodeInBranchPath of branch.nodesInBranchPath) {
      const nodeId = nodeInBranchPath.nodeId;
```